src/js/settings.js

```diff
@@ -506,7 +506,61 @@ function importSubscriptions(){
     })
   });
 }
+/**
+* Import NewPipe Channels Subscriptions
+* @return {Void}
+*/
+function importNewpipeSubscriptions(){
+  // Open user's file browser.  Only show .json files.
+ dialog.showOpenDialog({
+  properties: ['openFile'],
+  filters: [
+    {name: 'JSON', extensions: ['*']},
+  ]
+}, function(fileLocation){
+  if(typeof(fileLocation) === 'undefined'){
+    console.log('Import Aborted');
+    return;
+  }
+  console.log(fileLocation);
+  let i = fileLocation[0].lastIndexOf('.');
+  let fileType = (i < 0) ? '' : fileLocation[0].substr(i);
+  console.log(fileType);
+
+  fs.readFile(fileLocation[0], function(readErr, data){
+    if(readErr){
+      showToast('Unable to read file.  File may be corrupt or have invalid permissions.');
+      throw readErr;
+    }
 
+    if (data.includes("app_version")){
+      let newpipe, n, link, newpipesub;
+      
+      newpipe = JSON.parse(data);
+      
+      for (n in newpipe.subscriptions) {
+
+          link = newpipe.subscriptions[n].url.split("/");
+  
+          invidiousAPI('channels', link[4], {}, (data)=> {
+            newpipesub = {
+              channelId: data.authorId,
+              channelName: data.author,
+              channelThumbnail: data.authorThumbnails[2].url
+            };
+
+            showToast("Importing Newpipe Subscriptions. Please restart freetube when this message disappears, to see changes.");
```

Comment:
FreeTube will not require a restart when importing in this way.  You can remove this line.